ProfileManager.js

```diff
@@ -126,24 +124,23 @@ export default class ProfileManager {
   }
 
   /**
-   * Gets the profile agent assigned to the account associated with the
-   * authenticated session for the profile identified by the given profile ID.
+   * Gets a capability specified by its reference ID for profile agent from
+   * a cache or delegates one on the fly if it does not exist.
    *
    * @param {object} options - The options to use.
-   * @param {string} options.profileId - The ID of the profile to get the
-   *   profile agent for.
-   * @param {string} options.capabilityInvoker - The type of invoker to sign
-   * capability invocations.
+   * @param {string} options.id - The reference ID of the zcap.
+   * @param {string} options.profileAgent - The profile agent requesting a zcap.
+   * @param {boolean} options.useEphemeralSigner - Flag to enable invoking
+   *  capabilities with the ephemeral invocation signer, default `true`.
    *
-   * @returns {Promise<object>} The profile agent.
+   * @returns {Promise<object>} The capability for the profile agent.
    */
-  async getCapability(
-    {id, profileAgent, profileId, capabilityInvoker = 'local'} = {}) {
+  async getAgentCapability({id, profileAgent, useEphemeralSigner = true}) {
     const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
-    const capabilityKey = `${capabilityCacheKey}-${id}-${capabilityInvoker}`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
 
-    if(!this._cache.has(capabilityCacheKey)) {
-      this._cache.set(capabilityCacheKey, new Cache());
+    if(!this._cacheContainer.has(capabilityCacheKey)) {
+      this._cacheContainer.set(capabilityCacheKey, new Cache());
     }
 
     let promise = this._requests.get(capabilityKey);
```

Comment:
Is there a reason why this `_requests` thing is here or can we just put a promise into the cache for the zcap and simplify like elsewhere?

Reply:
Removed and simplified.